Makefile

```diff
@@ -108,85 +123,114 @@ all: build
 # More info on the awk command:
 # http://linuxcommand.org/lc3_adv_awk.php
 
+.PHONY: help
 help: ## Display this help.
 	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
 
 ##@ Development
-
+.PHONY: manifests
 manifests: controller-gen ## Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
 	$(CONTROLLER_GEN) $(CRD_OPTIONS) rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
 
 generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
+.PHONY:generate
 	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
 
-fmt: goimports ## Run go goimports against code.
+.PHONY: fmt
+fmt: goimports ## Run go goimports against code - goimports = go fmt + fixing imports.
 	$(GOIMPORTS) -w ./api ./controllers ./test
 
-vet: ## Run go vet against code.
+.PHONY: vet
+vet: ## Run go vet against code - report likely mistakes in packages.
 	go vet ./api/... ./controllers/... ./test/...
 
-go-vendor:
+.PHONY: go-tidy
+go-tidy: # Run go mod tidy - add missing and remove unused modules.
+	go mod tidy
+
+.PHONY: go-vendor
+go-vendor:  # Run go mod vendor - make vendored copy of dependencies.
 	go mod vendor
 
-go-tidy:
-	go mod tidy
+.PHONY: go-verify
+go-verify: go-tidy go-vendor # Run go mod verify - verify dependencies have expected content
+	go mod verify
+
+.PHONY: test
+test: test-no-verify ## Generate and format code, run tests, generate manifests and bundle, and verify no uncommitted changes
```

Comment:
I'd expect that after refactoring, previous `test` would be logically equivalent to the new `test` target,
(i.e  old test would be broken into `test-no-verify` + `verify-changes`) but looks like this is not the case.
can you explain the reason for the logical changes ?

Reply:
That was an incorrect attempt. I have updated it according to what you have suggested,